Fix: replaced Files.readString with BufferedReader to avoid OOM (#41)

LeX1Ka opened the pull request:
Fixes #41
Replaced Files.readString with streaming via BufferedReader in collectRawLogs().
Prevents OutOfMemoryError when reading large log files.

samuelowino commented:
@LeX1Ka we need to validate test coverage before merging

Please run:

```bash
mvn verify
```

This will generate the latest test coverage report in project readme. A coverage of **100% is required** for this branch to be eligible for merge.

To identify any missing branches or lines:

1. Open the following file in your browser:

   ```
   /target/site/jacoco/index.html
   ```
2. Review the highlighted sections in the report to locate untested code.

📘 Reference guide for generating and interpreting reports:
👉 https://www.codingshuttle.com/spring-boot-handbook/ja-co-co-test-report-generation/

Once coverage is verified at 100%, commit the updated report and proceed with the merge.

samuelowino commented on src/main/java/com/kenyajug/regression/utils/DateTimeUtils.java:
```diff
@@ -151,15 +152,13 @@ public static String localDateTimeToUTCTime(LocalDateTime localDateTime) {
-            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX");
-            var temporal = formatter.parse(timestamp);
-            return LocalDateTime.from(temporal);
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MMM-yyyy HH:mm:ss.SSS", Locale.ENGLISH);
+            return LocalDateTime.parse(timestamp, formatter);
+        } catch (DateTimeParseException formatException) {
+            DateTimeFormatter fallbackFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSSX");
+            return LocalDateTime.parse(timestamp, fallbackFormatter);
```
How does this affect the issue this PR is meant to address? Logs loading performance

samuelowino reviewed (changes requested):
Don't expand the scope of the PR

samuelowino commented on target/site/jacoco/index.html:
```diff
@@ -0,0 +1 @@
+<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="ru"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="jacoco-resources/re ...
```
The test coverage is automatically written to the project readme, you should not create this file

Once you run `mvn verify` the readme file should automatically update

Consider deleting this file

samuelowino commented:
@LeX1Ka this is the test coverage report I'm referring to, you should find it in the project readme

<img width="940" alt="Screenshot 2025-05-20 at 23 08 06" src="https://github.com/user-attachments/assets/a57934b3-2952-4f99-8c7e-81f0d98bf8b7" />

samuelowino commented:
> I’ve restored the DateTimeUtils file and cleaned up the staged changes as requested. The README.md was also updated and committed.

Your commit is missing restoration of DateTimeUtils file and the removal of the html file. You probably forgot to add them to staged before committing. [cec2f8b](https://github.com/kenya-jug/regression/pull/59/commits/cec2f8bec69d87eb3c03e4009cff132aa0ac5d3c) only contains changes to the README file

samuelowino commented:
<img width="940" alt="Screenshot 2025-05-20 at 23 53 48" src="https://github.com/user-attachments/assets/6f44d1b0-ff7c-4616-a146-f5f48f49e627" />

LeX1Ka commented:
I removed the target/site/jacoco/index.html file from the repository, as it shouldn’t be committed.
Also, I updated the README to reflect this change.

Sorry if I missed something or if anything is still not perfect.
Thank you very much for your help and patience.

samuelowino commented:
> I removed the target/site/jacoco/index.html file from the repository, as it shouldn’t be committed. Also, I updated the README to reflect this change.
> 
> Sorry if I missed something or if anything is still not perfect. Thank you very much for your help and patience.

Brilliant fixes! The read and index.html are now removed.

The DateTimeUtils.java is still not resolved. 
You can fix this by copy pasting [this version](https://github.com/kenya-jug/regression/blob/main/src/main/java/com/kenyajug/regression/utils/DateTimeUtils.java) to your local DateTimeUtils.java file: https://github.com/kenya-jug/regression/blob/main/src/main/java/com/kenyajug/regression/utils/DateTimeUtils.java
After pasting the main repo version the diff should disappear

This will ensure your PR is clean

samuelowino reviewed (approved):
Looks good to go! Nice work 👍🏿
